### PR TITLE
Feature/september release ialuj

### PR DESCRIPTION
**Pull request by ialuj**
20th Jan Release Features PR

### CODE REVIEW

**Comment by ialuj** [comment]
New PR For 20th Jan Features

File: app/src/main/java/com/muzima/view/initialwizard/GuidedConfigurationWizardActivity.java [hunk 1]
```diff
@@ -764,6 +764,7 @@ protected void onPostExecute(int[] result) {
                 downloadConceptsLog.setSetupActionResultStatus(resultStatus);
                 onQueryTaskFinish();
                 if(!isOnlineOnlyModeEnabled) {
+                    downloadEncounters();
```
Comment (benamonya) [inline app/src/main/java/com/muzima/view/initialwizard/GuidedConfigurationWizardActivity.java:767]: We are not making use of the encounters. Please remove the download

File: app/src/main/java/com/muzima/view/patients/DataCollectionActivity.java [hunk 2]
```diff
@@ -347,12 +347,10 @@ private void loadPatientData() {
             patient = ((MuzimaApplication) getApplicationContext()).getPatientController().getPatientByUuid(patientUuid);
             patientNameTextView.setText(patient.getDisplayName());
             identifierTextView.setText(String.format(Locale.getDefault(), "ID:#%s", patient.getIdentifier()));
-            if (patient.getBirthdate() != null)
-                dobTextView.setText(getString(R.string.general_date_of_birth ,String.format(" %s", new SimpleDateFormat("MM-dd-yyyy", Locale.getDefault()).format(patient.getBirthdate()))));
-
-            patientGenderImageView.setImageResource(getGenderImage(patient.getGender()));
```
Comment (benamonya) [inline app/src/main/java/com/muzima/view/patients/DataCollectionActivity.java:353]: Why did you remove the gender Icon

File: app/src/main/java/com/muzima/view/patients/ObsViewActivity.java [hunk 3]
```diff
@@ -199,13 +199,11 @@ private void loadPatientData() {
...
-                dobTextView.setText(getString(R.string.general_date_of_birth ,String.format(": %s", new SimpleDateFormat("MM-dd-yyyy", Locale.getDefault()).format(patient.getBirthdate()))));
-
+            if (patient.getBirthdate() != null) {
+                ageTextView.setText(getString(R.string.general_years, String.format(Locale.getDefault(), "%d ", DateUtils.calculateAge(patient.getBirthdate()))));
+                dobTextView.setText(getString(R.string.general_date_of_birth, String.format(": %s", new SimpleDateFormat("dd-MM-yyyy", Locale.getDefault()).format(patient.getBirthdate()))));
+            }
             patientGenderImageView.setImageResource(getGenderImage(patient.getGender()));
-            if (patient.getBirthdate() != null)
-                ageTextView.setText(getString(R.string.general_years ,String.format(Locale.getDefault(), "%d ", DateUtils.calculateAge(patient.getBirthdate()))));
-            gpsAddressTextView.setText(getDistanceToClientAddress(patient));
```
Comment (benamonya) [inline app/src/main/java/com/muzima/view/patients/ObsViewActivity.java:208]: Why did you remove the gps address text.

File: app/src/main/java/com/muzima/view/patients/PatientSummaryActivity.java [hunk 4]
```diff
@@ -230,38 +259,40 @@ private void navigateToClientsLocationMap() {
...
             patientNameTextView.setText(patient.getDisplayName());
             identifierTextView.setText(String.format(Locale.getDefault(), "ID:#%s", patient.getIdentifier()));
-            if (patient.getBirthdate() != null)
-                dobTextView.setText(getString(R.string.general_date_of_birth ,String.format(" %s", new SimpleDateFormat("MM-dd-yyyy", Locale.getDefault()).format(patient.getBirthdate()))));
 
-            patientGenderImageView.setImageResource(getGenderImage(patient.getGender()));
-            if (patient.getBirthdate() != null)
+            if (patient.getBirthdate() != null) {
                 ageTextView.setText(getString(R.string.general_years, String.format(Locale.getDefault(), "%d ", DateUtils.calculateAge(patient.getBirthdate()))));
-            gpsAddressTextView.setText(getDistanceToClientAddress(patient));
```
Comment (benamonya) [inline app/src/main/java/com/muzima/view/patients/PatientSummaryActivity.java:257]: Why did you remove this line

File: app/src/main/java/com/muzima/view/patients/PatientSummaryActivity.java [hunk 5]
```diff
@@ -310,30 +529,55 @@ private String getObsByPatientUuidAndConceptId(String patientUuid, int conceptId
...
             Log.e(getClass().getSimpleName(), "Exception occurred while loading observations", e);
         }
         return StringUtils.EMPTY;
     }
 
+    private Observation getObservationByPatientUuidAndConceptId(String patientUuid, int conceptId) throws JSONException, ObservationController.LoadObservationException {
+        List<Observation> observations = new ArrayList<>();
+        try {
+            observations = observationController.getObservationsByPatientuuidAndConceptId(patientUuid, conceptId);
+            Concept concept = conceptController.getConceptById(conceptId);
```
Comment (benamonya) [inline app/src/main/java/com/muzima/view/patients/PatientSummaryActivity.java:554]: the concept is not used in the function, should be removed

File: app/src/main/res/values-gu/strings.xml [hunk 6]
```diff
@@ -790,4 +790,17 @@
     <string name="hint_network_connection_failure_map_loading">ઉપકરણ ઇન્ટરનેટથી કનેક્ટ થયેલ નથી. ઑફલાઇન હોય ત્યારે નકશો ઉપલબ્ધ નથી.</string>
     <string name="warning_new_version_available">mUzima (v%s) નું નવું સંસ્કરણ ઉપલબ્ધ છે.\n\nતમે અપડેટ ડાઉનલોડ કરતી વખતે mUzima નો ઉપયોગ કરવાનું ચાલુ રાખી શકો છો. એકવાર ડાઉનલોડ પૂર્ણ થઈ જાય પછી તમને સૂચિત કરવામાં આવશે અને નવું સંસ્કરણ ઇન્સ્ટોલ કરવાની પરવાનગી માટે પૂછવામાં આવશે. વિનંતી કર્યા પછી કૃપા કરીને પરવાનગી આપો.\n\nઅપડેટ કરવા આગળ વધીએ?</string>
     <string name="general_new_version">નવી આવૃત્તિ</string>
+    <string name="confidant_details">Detalhes do Confidente</string>
```
Comment (benamonya) [inline app/src/main/res/values-gu/strings.xml:793]: Translate to Gujarati using google translate

File: app/src/main/res/values-hi/strings.xml [hunk 7]
```diff
@@ -789,4 +789,17 @@
     <string name="hint_network_connection_failure_map_loading">डिवाइस इंटरनेट से कनेक्ट नहीं है। ऑफ़लाइन होने पर नक्शा उपलब्ध नहीं होता है।</string>
     <string name="warning_new_version_available">mUzima (v%s) का एक नया संस्करण उपलब्ध है।\n\nआप अपडेट को डाउनलोड करते समय mUzima का उपयोग जारी रख सकते हैं। एक बार डाउनलोड पूरा हो जाने पर आपको सूचित किया जाएगा और नए संस्करण को स्थापित करने की अनुमति मांगी जाएगी। कृपया अनुरोध करने के बाद अनुमति दें।\n\nअपडेट करने के लिए आगे बढ़ें?</string>
     <string name="general_new_version">नया संस्करण</string>
+    <string name="confidant_details">Detalhes do Confidente</string>
```
Comment (benamonya) [inline app/src/main/res/values-hi/strings.xml:792]: Translate to Hindu using google translate

File: app/src/main/res/values-sw/strings.xml [hunk 8]
```diff
@@ -794,4 +794,17 @@
     <string name="hint_network_connection_failure_map_loading">Kifaa hakijaunganishwa kwenye mtandao. Ramani haipatikani ukiwa nje ya mtandao.</string>
     <string name="warning_new_version_available">Toleo jipya la mUzima (v%s) linapatikana.\n\nUnaweza kuendelea kutumia mUzima unapopakua sasisho. Upakuaji ukikamilika utaarifiwa na kuombwa ruhusa ya kusakinisha toleo jipya. Tafadhali toa ruhusa baada ya ombi lako.\n\nUngependa kusasisha?</string>
     <string name="general_new_version">Toleo Jipya</string>
+    <string name="confidant_details">Detalhes do Confidente</string>
```
Comment (benamonya) [inline app/src/main/res/values-sw/strings.xml:797]: Translate to Swahili using google translate

File: app/src/main/java/com/muzima/controller/MuzimaSettingController.java [hunk 9]
```diff
@@ -42,34 +42,7 @@
...
-import static com.muzima.util.Constants.ServerSettings.RELATIONSHIP_FEATURE_ENABLED;
-import static com.muzima.util.Constants.ServerSettings.NOTIFICATION_FEATURE_ENABLED_SETTING;
-import static com.muzima.util.Constants.ServerSettings.SHR_FEATURE_ENABLED_SETTING;
-import static com.muzima.util.Constants.ServerSettings.DEMOGRAPHICS_UPDATE_MANUAL_REVIEW_REQUIRED_SETTING;
-import static com.muzima.util.Constants.ServerSettings.SINGLE_ELEMENT_ENTRY_FEATURE_ENABLED_SETTING;
-
-import static com.muzima.util.Constants.ServerSettings.TAG_GENERATION_ENABLED_SETTING;
-
-import static com.muzima.util.Constants.ServerSettings.ADD_CONTACT_POPUP_ENABLED_SETTING;
+import static com.muzima.util.Constants.ServerSettings.*;
```
Comment (benamonya) [inline app/src/main/java/com/muzima/controller/MuzimaSettingController.java:45]: Import individual imports and not *

File: app/src/main/java/com/muzima/view/patients/PatientSummaryActivity.java (hunk 5 above)
Comment (ialuj) [reply app/src/main/java/com/muzima/view/patients/PatientSummaryActivity.java]: Already solved on the last commit and instead of picking up the concept from conceptController... Picking it up from the obs object.

**Comment by ialuj** [comment]
All conflicting conflicts were resolved and for the date format we need to display it as **dd-MM-yyyy** - day, Month and Year.

File: app/src/main/java/com/muzima/view/patients/PatientSummaryActivity.java [hunk 10]
```diff
@@ -521,10 +521,10 @@ private String getFormattedPatientAddress(PersonAddress personAddress) {
     private String getObsByPatientUuidAndConceptId(String patientUuid, int conceptId) throws JSONException, ObservationController.LoadObservationException {
         try {
             List<Observation> observations = observationController.getObservationsByPatientuuidAndConceptId(patientUuid, conceptId);
-            Concept concept = conceptController.getConceptById(conceptId);
             Collections.sort(observations, observationDateTimeComparator);
             if (observations.size() > 0) {
                 Observation obs = observations.get(0);
+                Concept concept = obs.getConcept();
                 if (concept.isDatetime())
```
Comment (benamonya) [inline app/src/main/java/com/muzima/view/patients/PatientSummaryActivity.java:528]: You cant get .isDateTime from concept retrieved from obs, instead do this Concept concept = conceptController.getConceptById(conceptId);

**Review by benamonya — APPROVED** [review]
